lib/spack/spack/mixins.py

```diff
@@ -0,0 +1,177 @@
+        """
+        PackageMixinsMeta._methods_to_be_added[fn.__name__] = fn
+        PackageMixinsMeta._add_method_after[phase].append(fn)
+
+    def __init__(cls, name, bases, attr_dict):
+
+        # Add the methods to the class being created
+        if PackageMixinsMeta._methods_to_be_added:
+            attr_dict.update(PackageMixinsMeta._methods_to_be_added)
+            PackageMixinsMeta._methods_to_be_added.clear()
+
+        attr_fmt = '_InstallPhase_{0}'
+
+        # Copy the phases that needs it to the most derived classes
+        # in order not to interfere with other packages in the hierarchy
+        phases_to_be_copied = list(
+            PackageMixinsMeta._add_method_before.keys()
+        )
+        phases_to_be_copied += list(
+            PackageMixinsMeta._add_method_after.keys()
+        )
+
+        for phase in phases_to_be_copied:
+
+            attr_name = attr_fmt.format(phase)
+
+            # Here we want to get the attribute directly from the class (not
+            # from the instance), so that we can modify it and add the mixin
+            # method to the pipeline.
+            phase = getattr(cls, attr_name)
+
+            # Due to MRO, we may have taken a method from a parent class
+            # and modifying it may influence other packages in unwanted
+            # manners. Solve the problem by copying the phase into the most
+            # derived class.
+            setattr(cls, attr_name, phase.copy())
+
+        # Insert the methods in the appropriate position
+        # in the installation pipeline.
+
+        for phase in PackageMixinsMeta._add_method_before:
+
+            attr_name = attr_fmt.format(phase)
+            phase_obj = getattr(cls, attr_name)
+            fn_list = PackageMixinsMeta._add_method_after[phase]
+
+            for f in fn_list:
+                phase_obj.run_before.append(f)
+
+        # Flush the dictionary for the next class
+        PackageMixinsMeta._add_method_before.clear()
+
+        for phase in PackageMixinsMeta._add_method_after:
+
+            attr_name = attr_fmt.format(phase)
+            phase_obj = getattr(cls, attr_name)
+            fn_list = PackageMixinsMeta._add_method_after[phase]
+
+            for f in fn_list:
+                phase_obj.run_after.append(f)
+
+        # Flush the dictionary for the next class
+        PackageMixinsMeta._add_method_after.clear()
+
+        super(PackageMixinsMeta, cls).__init__(name, bases, attr_dict)
+
+
+def filter_compiler_wrappers(*files, **kwargs):
+    """Substitutes any path referring to a Spack compiler wrapper with the
+    path of the underlying compiler that has been used.
+
+    If this isn't done, the files will have CC, CXX, F77, and FC set to
+    Spack's generic cc, c++, f77, and f90. We want them to be bound to
+    whatever compiler they were built with.
+
+    Args:
+        *files: files to be filtered
+        **kwargs: at present supports the keyword 'after' to specify after
```

Comment:
The fact that this parameter is a a `**kwargs` is mandated by the previous request of having the first parameter be a `*varargs`. To be more precise, what you ask is possible in `python@3`, but not in `python@2`. References [here](https://docs.python.org/2/reference/compound_stmts.html#function-definitions) and [here](https://docs.python.org/3.6/reference/compound_stmts.html#function-definitions) for your convenience.

---

Bottomline: here I can either turn the first parameter to be an iterable, and have an explicit keyword argument after it, or leave things like they are. Let me know which solution you prefer.